### ISSUE TITLE
[Validator] Load the methodMappings after the annotations

### ISSUE THREAD

**Issue by develth** [symfony/symfony#24299]
I want to achieve that my Serializer Groups are set to the Validation Constraint, so i don´t have to  add lot of same lines in my annotations. 

Currently the MethodMappings call in the ValidationBuilder is  before the AnnotationLoader, so if i pass the metaData into my Class to validate, its empty (i´m  just using annotation validation).
By moving it as last addition to the loader array, it is would be possible to handle the metaData dynamically with e.g.

```php
    public static function loadValidatorMetadata(ClassMetadata $metaData){
        $props = self::getSerializerProperties(self::class);
        if(count($props) > 0){
           foreach($metaData->properties as $propertyMetadata){
               if($props->available($propertyMetadata->getName(), "groups")){
                   foreach ($propertyMetadata->constraints as $constraint){
                       $constraint->groups = $props->get($propertyMetadata->getName(), "groups")
                       $propertyMetadata->addConstraint($constraint);
                  }
               }
           }
        }
    }
```

**Comment by develth** [comment]
fixed with #24298

**Comment by nicolas-grekas** [comment]
Closing as a better approach has been suggested in the linked PR.
@develth feel free to submit an issue or a PR to keep track of this if you want to to happen.
